File: elixir/bob/bob.exs
defmodule Teenager do
  def hey(input) do
    cond do
      String.upcase(input) == input ->
        "Woah, chill out!"
      String.at(input, -1) == "?" ->
        "Sure."
      true ->
        "Whatever."
    end
  end
end

File: fsharp/bob/Bob.fs
module Bob

open System
open System.Linq

let hey (input : string) =
    match input with
      | input when input.Trim() = String.Empty -> "Fine. Be that way!"
      | input when (input.Any(fun x -> System.Char.IsLetter x)) && input.ToUpper() = input -> "Whoa, chill out!"
      | input when input.EndsWith("?") -> "Sure."
      | input -> "Whatever."